### PR TITLE
feat: mark files as replaced

### PR DESCRIPTION
**Pull request by MitchellPaff**
## Reference

[Code review checklist](CODING.md#Code-review-checklist)

### CODE REVIEW

File: geostore/check_stac_metadata/utils.py [hunk 1]
```diff
@@ -150,7 +156,7 @@ def process_assets(self) -> None:
             assert isinstance(asset_url, str)
 
             self.processing_assets_model(
-                hash_key=self.hash_key,
+                hash_key=get_hash_key(self.dataset_id, self.new_version_id),
```
Comment (l0b0) [inline geostore/check_stac_metadata/utils.py:159]: Looks like `get_hash_key` is called for each metadata file and asset. We probably want to call it in the constructor instead.

File: geostore/step_function.py [hunk 2]
```diff
@@ -180,3 +181,29 @@ def get_hash_key(dataset_id: str, dataset_version_id: str) -> str:
...
+    items = processing_assets_model.query(
+        get_hash_key(dataset_id, current_version_id),
+        range_key_condition=processing_assets_model.sk.startswith(
+            f"{processing_asset_type.value}{DB_KEY_SEPARATOR}"
+        ),
+        filter_condition=(processing_assets_model.filename == filename),
+        consistent_read=True,
+    )
+
+    assert items.total_count > 1, "More than 1 file returned with the same filename"
```
Comment (l0b0) [inline geostore/step_function.py:205]: Did you mean to `assert items.total_count == 1`?

Also, the assertion message should probably include the relevant parameters to debug it, basically the dataset ID, version ID and filename.

File: geostore/step_function.py [hunk 3]
```diff
@@ -180,3 +181,29 @@ def get_hash_key(dataset_id: str, dataset_version_id: str) -> str:
...
+
+    if items.total_count == 1:
```
Comment (l0b0) [inline geostore/step_function.py:207]: This shouldn't be necessary with the assertion above.

File: geostore/step_function.py (hunk 2 above)
Comment (MitchellPaff) [reply geostore/step_function.py:205]: No, I meant to assert that total is  <2. 

It's a valid case that this query returns 0 - this is the case when this file is new in this version.

**Review by billgeo — APPROVED** [review]
Commits so far look good. But looks like a couple of tests are failing?

**Review by billgeo — APPROVED** [review]

File: tests/test_check_stac_metadata.py [hunk 4]
```diff
@@ -694,6 +698,72 @@ def should_successfully_validate_partially_uploaded_dataset(subtests: SubTests)
                     )
 
 
+@mark.timeout(timedelta(minutes=20).total_seconds())
+@mark.infrastructure
+def should_mark_asset_as_replaced_in_new_version() -> None:
```
Comment (l0b0) [inline tests/test_check_stac_metadata.py:703]: There are two tests with the exact same name. Based on #1779 I assume that this test was not meant to be here.

**Review by l0b0 — DISMISSED** [review]

**Review by l0b0 — APPROVED** [review]

**Pull request merged**